### PR TITLE
Divides the module into two parts (and more)

### PR DESCRIPTION
**Pull request by dpshelio**
This course is now going to be taught as two (consecutive) courses. This PR brings that in!

This is done by adding two sections on the main page and on the navbar
(content is now separated into two directories)

But it also brings some new stuff:
- Uses Jekyll remote-themes (to get updates directly from the theme)
- Updates to the latest version of jekyll
- Uses a new md parser (kramdown instead of redcap)
- pip packages should be also cached now (only the latex/pandoc stuff is missing now)
- Adds license to the repository!

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -16,75 +15,70 @@ you will be able to apply this knowledge to your
...
+  - The IPython Notebook
+  - Data structures with lists and dictionaries
+  - Files and the filesystem
+* [Part II](02-novice/)
+  - Parsing CSV data
+  - JSON and YAML
+  - Getting data from the internet
+  - Introducing Matplotlib
+  - Introducing NumPy
+  - Creating packages
```
Comment (matt-graham) [inline index.md:33]: This seems to differ a bit from the split we have in the HackMD planning page inasmuch that I think the plan was to have a basic intro to Matplotlib in the beginner course as well as a further lesson on more advanced usage of Matplotlib in the follow up?

File: index.md [hunk 2]
```diff
@@ -2,8 +2,7 @@
 title: An introduction to programming for research using python
```
Comment (matt-graham) [inline index.md:2]:
```suggestion
title: An introduction to programming for research using Python
```

File: 02-novice/index.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+title: Part II
+---
+## {{ page.title }}
+
+This is the second part of the Introduction to research programming with Python course
+
+This course introduces the basics of creating and interacting with scientific Python libraries such as numpy, scipy and matplotlib. If you are familiar with the basics of programming (variables, loops and functions) then this course offers you the next step – how to structure your code and organise it into modules.
```
Comment (matt-graham) [inline 02-novice/index.md:8]:
```suggestion
This course introduces the basics of creating and interacting with scientific Python libraries such as [NumPy](https://numpy.org/, [SciPy](https://scipy.org/) and [Matplotlib](https://matplotlib.org/). If you are familiar with the basics of programming (variables, loops and functions) then this course offers you the next step – how to structure your code and organise it into modules.
```

File: 02-novice/index.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+title: Part II
+---
+## {{ page.title }}
+
+This is the second part of the Introduction to research programming with Python course
```
Comment (matt-graham) [inline 02-novice/index.md:6]:
```suggestion
This is the second part of the *Introduction to research programming with Python* course
```

File: 01-beginner/index.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+2. The different ways to use Python
+3. Data structures with lists and dictionaries
+4. Repeating actions and making choices
+5. Files and the filesystem
+6. Making functions
+7. Basic data visualisation
+    
+### By the end of this course you will be able to:
+
+1. Use python from different interfaces
```
Comment (matt-graham) [inline 01-beginner/index.md:25]:
```suggestion
1. Use Python from different interfaces
```

File: 01-beginner/index.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+---
+title: Part I
+---
+## {{ page.title }}
+
+This is the first part of the Introduction to research programming with Python course.
```
Comment (matt-graham) [inline 01-beginner/index.md:6]:
```suggestion
This is the first part of the *Introduction to research programming with Python* course.
```

File: 01-beginner/index.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+This course teaches the basics of programming using Python with the focus to use
+it for research. It’s aimed to researchers that have no experience of
```
Comment (matt-graham) [inline 01-beginner/index.md:9]:
```suggestion
it for research. It’s aimed at researchers that have no experience of
```

File: 01-beginner/index.md (hunk 6 above)
Comment (matt-graham) [reply 01-beginner/index.md:6]: (putting this in italics I thinks makes it more obvious that the capitalization is because this is the course name)

**Review by matt-graham — CHANGES_REQUESTED** [review]
Looks great in general. I've made a few small suggestions - mainly changing capitalization and adding a bit of formatting. The main thing I think needs potentially changing is that my understanding was that the `notebooks/060MatplotlibPyplot.ipynb` notebook should be part of the beginners course (as this is the new introductory notebook only covering the basics of using the `pyplot` interface) and so moved to the `01-beginner` subdirectory rather than `02-novice`? Based on our last in person meeting I think we'd said we would include a basic intro to Matplotlib in the beginners course, but cover the object oriented interface and more advanced usage in the second course.

File: index.md [hunk 8]
```diff
@@ -16,75 +15,70 @@ you will be able to apply this knowledge to your
...
+  - Getting data from the internet
+  - Introducing Matplotlib
+  - Introducing NumPy
+  - Creating packages
+
+## Alternatives
 
 If you're already comfortable, with variables, loops, and functions, this course is not for you!
-You should join us instead for [MPHYG001: Research Software Engineering with Python](http://github-pages.ucl.ac.uk/rsd-engineeringcourse/)
+You should join us instead for [MPHYG021: Research Software Engineering with Python](http://github-pages.ucl.ac.uk/rsd-engineeringcourse/)
```
[the remark this answers is not in the document]
Comment (dpshelio) [reply index.md:38]: That was a remanent from when `MPHY0021` was `MPHYG001` - I never knew what the `G` was for :sweat_smile:

**Review by matt-graham — APPROVED** [review]

**Pull request merged**